doc/contrib/coding-style.rst

```diff
@@ -3,6 +3,26 @@
 Coding Guidelines
 =================
 
+Python
+------
+
+We follow the `black <https://black.readthedocs.io/en/stable/index.html>`__
+coding style. It is enforced by an automated check on each pull request. You can
+run the following commands to apply it:
+
+.. code::
+
+   # Install the formatter if not present
+   pip install black
+   # Automatically apply style. If unsure what this does read on.
+   black --extend-exclude '/(ext|python/pybind11|doc/scripts/.*_theme)' . scripts/build-catalogue.in
+
+The formatter can also be run with ``--check`` to list offending files and
+``--diff`` to preview changes.
```

Comment:
Let's mention how people can set their editor to do this for them: https://black.readthedocs.io/en/stable/integrations/editors.html

Reply:
👍